tests/core/test_core.py

```diff
@@ -467,6 +467,34 @@ def test_stream_muting_confirmation_popup(
         )
         pop_up.assert_called_once_with(controller, text(), partial())
 
+    @pytest.mark.parametrize(
+        "search_text, expected_search_text, expected_search_narrow",
+        [
+            ("FOO", "FOO ", []),
+            ("FOO stream:PTEST", "FOO ", [["stream", "PTEST"]]),
+            (
+                "FOO stream:PTEST topic:TEST",
+                "FOO ",
+                [["stream", "PTEST"], ["topic", "TEST"]],
+            ),
+            ("FOO is:private", "FOO ", [["is", "private"]]),
+            ("FOO is:mentioned", "FOO ", [["is", "mentioned"]]),
+            ("FOO is:starred", "FOO ", [["is", "starred"]]),
```

Comment:
It'd be great to see some 'space-filled' examples here, like with the other function tests.